src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseStatusSearchHandler.java

```diff
@@ -42,18 +47,24 @@ public boolean canHandle(
         }
 
         final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
-        final Optional<String> homeOfficeReferenceNumber = asylumCase.read(
-            AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER);
+        final String homeOfficeReferenceNumber = asylumCase.read(HOME_OFFICE_REFERENCE_NUMBER, String.class)
+            .orElseThrow(() -> new IllegalStateException("Home office reference for the appeal is not present"));
 
-        if (!homeOfficeReferenceNumber.isPresent()) {
-            throw new IllegalStateException("Home office reference number cannot be null");
-        }
+        HomeOfficeSearchResponse searchResponse = homeOfficeSearchService.getCaseStatus(homeOfficeReferenceNumber);
+        requireNonNull(searchResponse.getStatus().get(0).getPerson(),
```

Comment:
Could it throw IndexOutOfBoundException here?

Reply:
The response should contain a status[] array and it could be an empty array. Yes, it could throw that exception

Reply:
Checked for all elements and set status that we can use later in the flow